packages/scm/src/browser/scm-avatar-service.ts

```diff
@@ -18,7 +18,7 @@ import { injectable } from 'inversify';
 import { Md5 } from 'ts-md5';
 
 @injectable()
-export class GitAvatarService {
+export class ScmAvatarService {
```

Comment:
Is there a reason this has been moved to SCM? I think it belongs into the concrete scm implementation (i.e. Git)

Reply:
The avatar service was moved into the SCM package because it is needed when rendering the last commit and the list of commits being amended.  As the SCM package cannot depend on the Git package, leaving the avatar service in the Git package would have required API to be defined to get an avatar from a concrete implementation.

The avatar service gets an avatar using only an e-mail address and has no dependency on Git.  Furthermore no SCM that I know of provides avatars, they all just provide an e-mail address which can then be used with a generic service to get an avatar.  So it does not make sense to me to put the avatar service in a concrete SCM implementation - it would just be duplicated identically in every one.  (or worse, it would not be identically implemented in each one, leaving users with different avatars within the IDE depending on the SCM).

Also consider the case where an IDE provider re-binds the avatar service, perhaps to get photos from an internal corporate database.  That would not be possible if each SCM implementation decided how it was going to get avatars from the e-mail address.

Reply:
Ok, thanks for the explanation. Makes sense.